src/ray/gcs/redis_context.h

```diff
@@ -21,7 +21,7 @@ namespace gcs {
 
 class RedisCallbackManager {
  public:
-  using RedisCallback = std::function<void(const std::string &)>;
+  using RedisCallback = std::function<void(const std::string &, bool is_nil)>;
```

Comment:
Does it make sense to add this parameter for all `RedisCallbacks`? I'm a little wary of adding this when it is only used by one of the table operations.

An alternative plan would be to call the failure callback if the response is empty in the `Lookup` call. Then we can keep the changes completely local to the `Lookup` operation. The only drawback I see here is that if we won't be able to support an entry that has an empty string as the value, but I think this is probably okay.